specification/datamigration/resource-manager/Microsoft.DataMigration/stable/2018-04-19/definitions/Common.json

```diff
@@ -0,0 +1,310 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Database Migration Service Resource Provider",
+    "version": "2018-04-19"
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "ApiError": {
+      "type": "object",
+      "description": "Error information.",
+      "properties": {
+        "error": {
+          "description": "Error information in OData format",
+          "$ref": "#/definitions/ODataError"
+        }
+      }
+    },
+    "NameAvailabilityResponse": {
+      "type": "object",
+      "description": "Indicates whether a proposed resource name is available",
+      "properties": {
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "If true, the name is valid and available. If false, 'reason' describes why not."
+        },
+        "reason": {
+          "type": "string",
+          "description": "The reason why the name is not available, if nameAvailable is false",
+          "enum": [
+            "AlreadyExists",
+            "Invalid"
+          ],
+          "x-ms-enum": {
+            "name": "NameCheckFailureReason",
+            "modelAsString": true
```

Comment:
>"modelAsString": true [](start = 12, length = 21)

just a reminder - if enum is not modeled as string any time you will need to add a new enum member to it will be a breaking change and will require an API version bump. I'd strongly recommend marking enums as string

Reply:
Just confirming - There is no change needed right? The usage here follows the recommended way of defining enums, correct?

Reply:
Currently all the enums are marked as "modelAsString": true

Reply:
No changes asked for now, just a warning for a possible breaking change in the future.

Reply:
My comment rather came from already existing enums in Tasks.json (e.g. ProjectTaskProperties.State). Your PR has all the new enums marked as string types which is exactly what I was recommending :) sorry for the confusion.